crates/bevy_ecs/src/system/mod.rs

```diff
@@ -1,3 +1,46 @@
+//! The behavior building block of an ECS application.
+//!
+//! Systems define how an ECS based application behave. They have to be registered to a
+//! [`SystemStage`](crate::schedule::SystemStage) to be able to run. A system is usually written as a normal function,
+//! on which the [`.system()`](IntoSystem::system) trait extension method is used, to turn it
+//! into a Bevy system.
+//!
+//! System functions can have parameters, through which one can query and mutate Bevy ECS
+//! states. Some limitations on allowed parameters apply (See [`SystemParam`]).
+//!
+//! # System ordering
+//!
+//! While the execution of systems is usually parallel and not deterministic, there are two
+//! ways to determine a certain degree of execution order:
+//!
+//! - **System Stages:** They determine hard execution synchronization boundaries inside of
+//!   which systems run in parallel by default.
+//! - **Labeling:** First, systems are labeled upon creation by calling `.label()`. Then,
+//!   methods such as `.before()` and `.after()` are appended to systems to determine
+//!   execution order in respect to other systems.
+//!
+//! # System parameter list
+//! Following is the complete list of accepted types as system parameters:
+//!
+//! - All tuples between 1 to 16 elements where each element implements `SystemParam`
+//! - [`()` (unit primitive type)](https://doc.rust-lang.org/stable/std/primitive.unit.html)
+//! - [`Archetypes`](crate::archetype::Archetypes)
+//! - [`Bundles`](crate::bundle::Bundles)
+//! - [`Commands`]
+//! - [`Components`](crate::component::Components)
+//! - [`Entities`](crate::entity::Entities)
+//! - [`EventReader`](crate::event::EventReader)
+//! - [`EventWriter`](crate::event::EventWriter)
+//! - [`Local`]
+//! - [`NonSend`]
+//! - [`NonSendMut`]
+//! - `Option<Res>`
+//! - `Option<ResMut>`
+//! - [`Query`]
+//! - [`RemovedComponents`]
+//! - [`Res`]
+//! - [`ResMut`]
```

Comment:
#2345 added `Option<NonSend>` and `Option<NonSendMut>`.
Also moving the Option Variants to same line as the Base Forms seem better to me:
```suggestion
//! - [`NonSend`] and `Option<NonSend>`
//! - [`NonSendMut`] and `Option<NonSendMut>`
//! - [`Query`]
//! - [`RemovedComponents`]
//! - [`Res`] and `Option<Res>`
//! - [`ResMut`] and `Option<ResMut>`
```

Reply:
> Also moving the Option Variants to same line as the Base Forms seem better to me

Yes, totally worth it, thanks!